packages/legacy-client/src/client/queries.ts

```diff
@@ -125,7 +125,11 @@ export type QueryDataTypeBase<
   O extends OntologyDefinition<any>,
   T extends QueryDataType<O, any, R>,
   R extends boolean,
-> = T extends PrimitiveDataType<infer X> ? ValidLegacyBaseQueryDataTypes[X]
+> = T extends PrimitiveDataType<infer X>
+  // returned longs are always strings, but we can accept numbers as input
+  ? (R extends true
```

Comment:
What is the R generic for?

Reply:
And also, so we don't need to worry about decimals here right?

Reply:
R is for "is this a return type?"
`QueryDefinitions` don't allow Decimals 🤷